### PR TITLE
Default Audio Profile

### PR DESCRIPTION
**Pull request by ykeara**
1. Added default audio profiles
2. PTT and audio profiles now stay visible after you select them.

Known Issues

1. Does not work on linux or Mac(current design)
2. Does not apply Default profiles correctly if advanced settings is launched directly (launch steamVR/openVR first)
3. Mirror Volume can be "remembered" when switching to and from playback Device
4. Currently Profiles change the steam/openVR audio preferences (discussion needed on correct course for future update)

Misc Fixes.
Fixed issue with audio page introduced during refactor
Fixed minor formatting in documentation
changed compile target to c++14

### CODE REVIEW

**Comment by ykeara** [comment]
I will change master1/parent1 var names in my next pr

File: advancedSettings.pro [hunk 1]
```diff
@@ -5,7 +5,9 @@
 #-------------------------------------------------
 
 QT       += core gui qml quick multimedia widgets
-CONFIG   += c++11
+CONFIG   += c++14
+
+DEFINES += ELPP_THREAD_SAFE ELPP_QT_LOGGING ELPP_NO_DEFAULT_LOG_FILE
```
Comment (username223) [inline advancedSettings.pro]: `DEFINES += ELPP_THREAD_SAFE ELPP_QT_LOGGING ELPP_NO_DEFAULT_LOG_FILE` is already inserted below.

File: src/res/qml/AudioPage.qml [hunk 2]
```diff
@@ -558,6 +575,9 @@ MyStackViewPage {
             audioPttRightControllerToggle.checked = AudioTabController.pttRightControllerEnabled
             audioPttShowNotificationToggle.checked = AudioTabController.pttShowNotification
             audioPttReverseToggle.checked = AudioTabController.micReversePtt
+            audioDefaultProfileToggle.checked = AudioTabController.audioProfileDefault
+            //audioProfileComboBox.currentIndex = AudioTabController.audioProfileDefaultIndex + 1
+            //AudioTabController.applyDefaultProfile()
```
Comment (username223) [inline src/res/qml/AudioPage.qml]: Unexplained commented out code?

File: src/res/qml/AudioPage.qml [hunk 3]
```diff
@@ -629,12 +649,27 @@ MyStackViewPage {
...
             }
+            onAudioProfileAdded:{
+                audioProfileComboBox.currentIndex = AudioTabController.getAudioProfileCount()
+                //audioProfileComboBox.currentIndex = AudioTabController.audioProfileDefaultIndex+1
+            }
+            onPttProfileAdded:{
+                pttProfileComboBox.currentIndex = AudioTabController.getPttProfileCount()
+            }
+
+            //onAudioProfileDefaultIndexChanged: {
```
Comment (username223) [inline src/res/qml/AudioPage.qml]: Unexplained commented out code?

File: src/res/qml/AudioPage.qml [hunk 4]
```diff
@@ -629,12 +649,27 @@ MyStackViewPage {
...
+            //         audioProfileComboBox.currentIndex = AudioTabController.audioProfileDefaultIndex()
```
Comment (username223) [inline src/res/qml/AudioPage.qml]: Unexplained commented out code?

File: src/tabcontrollers/AudioTabController.cpp [hunk 5]
```diff
@@ -432,7 +453,7 @@ int AudioTabController::getPlaybackDeviceCount()
 
 QString AudioTabController::getPlaybackDeviceName( int index )
 {
-    if ( index > 0 && static_cast<size_t>( index ) < m_playbackDevices.size() )
+    if ( index >= 0 && static_cast<size_t>( index ) < m_playbackDevices.size() )
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.cpp]: Good catches finding my mistakes. 👍

File: src/tabcontrollers/AudioTabController.cpp [hunk 6]
```diff
@@ -726,7 +746,7 @@ void AudioTabController::reloadAudioProfiles()
     {
         settings->setArrayIndex( i );
         audioProfiles.emplace_back();
-        auto& entry = audioProfiles[static_cast<size_t>( i )];
+        auto& entry = audioProfiles[i];
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.cpp]: Removing static_cast.

File: src/tabcontrollers/AudioTabController.cpp [hunk 7]
```diff
@@ -761,7 +783,7 @@ void AudioTabController::saveAudioProfiles()
     auto settings = OverlayController::appSettings();
     settings->beginGroup( getSettingsName() );
     settings->beginWriteArray( "audioProfiles" );
-    int i = 0;
+    unsigned i = 0;
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.cpp]: Changing type for no reason.

File: src/tabcontrollers/AudioTabController.h [hunk 8]
```diff
@@ -71,6 +68,9 @@ class AudioTabController : public PttController
     bool m_micMuted = false;
     bool m_micProximitySensorCanMute = false;
     bool m_micReversePtt = false;
+    bool m_isDefaultAudioProfile = false;
+    // int m_defaultAudioProfileIndex = 2;
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.h]: Commented out code?

File: src/tabcontrollers/AudioTabController.h [hunk 9]
```diff
@@ -126,17 +125,19 @@ class AudioTabController : public PttController
     bool micMuted() const;
     bool micProximitySensorCanMute() const;
     bool micReversePtt() const;
+    bool audioProfileDefault() const;
+    // int audioProfileDefaultIndex() const;
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.h]: Introducing commented out code.

File: src/tabcontrollers/AudioTabController.h [hunk 10]
```diff
@@ -101,14 +100,14 @@ class AudioTabController : public PttController
...
+
+    void removeDefaultProfile( QString name );
 
     std::vector<AudioProfile> audioProfiles;
-    // std::recursive_mutex eventLoopMutexAudio;
 
 public:
     void initStage1();
-    void initStage2( OverlayController* parent, QQuickWindow* widget );
+    void initStage2( OverlayController* parent1, QQuickWindow* widget1 );
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.h]: Unnecessary rename.

File: src/tabcontrollers/AudioTabController.cpp [hunk 11]
```diff
@@ -51,18 +51,21 @@ void AudioTabController::initStage1()
...
 }
 
-void AudioTabController::initStage2( OverlayController* var_parent,
-                                     QQuickWindow* var_widget )
+void AudioTabController::initStage2( OverlayController* parent1,
+                                     QQuickWindow* widget1 )
 {
-    this->parent = var_parent;
-    this->widget = var_widget;
+    this->parent = parent1;
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.cpp]: Unnecessary rename.

File: src/tabcontrollers/AudioTabController.cpp [hunk 12]
```diff
@@ -920,17 +954,15 @@ QString AudioTabController::getAudioProfileName( unsigned index )
...
+Name: getPlaybackIndex,  getRecordingIndex, and getMirrorIndex
+input: string, of microphone/playback device name
+output: integer for use In: setMicDeviceIndex(int,bool),
+setMirrorDeviceIndex(int,bool) setPlayBackDeviceIndex(int,bool)
+description: Gets proper index value for selecting specific devices.
 */
 int AudioTabController::getPlaybackIndex( std::string str )
 {
-    // Unsigned to avoid two casts. i won't be negative because we only
+    // Unsigned to avoid two casts. it won't be negative because we only
```
Comment (username223) [inline src/tabcontrollers/AudioTabController.cpp]: Sentence was meant as in `[The variable] i won't be negative [...]`, but whatever. :D

File: src/tabcontrollers/AudioTabController.cpp (hunk 11 above)
Comment (ykeara) [reply src/tabcontrollers/AudioTabController.cpp]: yea caught that one planed to change it in future but I will hit it up.

File: advancedSettings.pro (hunk 1 above)
Comment (ykeara) [reply advancedSettings.pro]: fixed dunno how it happened

File: src/tabcontrollers/AudioTabController.cpp (hunk 5 above)
Comment (ykeara) [reply src/tabcontrollers/AudioTabController.cpp]: <3

File: src/tabcontrollers/AudioTabController.cpp (hunk 6 above)
Comment (ykeara) [reply src/tabcontrollers/AudioTabController.cpp]: cast is un-necessary, and int is better choice for optimization

File: src/tabcontrollers/AudioTabController.h (hunk 8 above)
Comment (ykeara) [reply src/tabcontrollers/AudioTabController.h]: removed

File: src/tabcontrollers/AudioTabController.cpp (hunk 7 above)
Comment (ykeara) [reply src/tabcontrollers/AudioTabController.cpp]: yep fixed

**Pull request merged**